cns/client_test.go

```diff
@@ -1180,6 +1180,26 @@ func TestClient(t *testing.T) {
 			}
 		}
 
+		// Test BackingDiskObjectId field
+		var queryFilterBackingDiskObjectIdTest cnstypes.CnsQueryFilter
+		var volumeIDListBackingDiskObjectIdTest []cnstypes.CnsVolumeId
+		volumeIDListBackingDiskObjectIdTest = append(volumeIDListBackingDiskObjectIdTest, cnstypes.CnsVolumeId{Id: volumeID})
+		queryFilterBackingDiskObjectIdTest.VolumeIds = volumeIDListBackingDiskObjectIdTest
+		t.Logf("Calling QueryVolume using queryFilter: %+v", pretty.Sprint(queryFilterBackingDiskObjectIdTest))
+		queryResultBackingDiskObjectIdTest, err := cnsClient.QueryVolume(ctx, queryFilterBackingDiskObjectIdTest)
+		if err != nil {
+			t.Errorf("Failed to query volume. Error: %+v \n", err)
+			t.Fatal(err)
+		}
+		t.Logf("Successfully Queried Volumes. queryResult: %+v", pretty.Sprint(queryResult))
+		t.Log("Checking backingDiskObjectId retieved")
+		for _, vol := range queryResultBackingDiskObjectIdTest.Volumes {
+			backingDiskObjectId := vol.BackingObjectDetails.(*cnstypes.CnsBlockBackingDetails).BackingDiskObjectId
+			if backingDiskObjectId == "" {
+				t.Errorf("Failed to get BackingDiskObjectId")
```

Comment:
Return an error is not quite right. VMFS and NFS don't have backingDiskObjectId while vSAN and vVol do. 

It might better to only error out volumes on vVol or vSAN don't have backingDiskObjectId.

Reply:
Added check for datastore type.